### PR TITLE
DexDrip

### PR DESCRIPTION
**Pull request by bhandfast**
Receive data from DexDrip as intents.

See https://github.com/StephenBlackWasAlreadyTaken/DexDrip/pull/44

### CODE REVIEW

File: mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/DexdripIntentService.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+        final String action = intent.getAction();
+
+        try {
+            if (ACTION_NEW_DATA.equals(action)) {
+                final double bgEstimate = intent.getDoubleExtra(DexDripIntents.EXTRA_BG_ESTIMATE, 0);
+                if (bgEstimate == 0)
+                    return;
+
+                int battery = (int) (100 * intent.getIntExtra(DexDripIntents.EXTRA_SENSOR_BATTERY, 0) / 255f);
```
Comment (StephenBlackWasAlreadyTaken) [inline mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/DexdripIntentService.java]: Ah, you do handle that here! Well excellent then!!

**Comment by StephenBlackWasAlreadyTaken** [comment]
This is super cool, trying it out now!

File: mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/DexdripDataReceiver.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package com.dexdrip.stephenblack.nightwatch.integration;
+
+import android.content.Context;
+import android.content.Intent;
+import android.support.v4.content.WakefulBroadcastReceiver;
+
+/**
+ * Tells us when G4 data has been updated.
+ */
+public class DexDripDataReceiver extends WakefulBroadcastReceiver {
```
Comment (StephenBlackWasAlreadyTaken) [inline mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/DexdripDataReceiver.java]: Due to the Drip being capitalized here but not in the file name, android studio complains about this class, mind adjusting one or the other?

File: mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/DexdripIntentService.java [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+import com.dexdrip.stephenblack.nightwatch.Bg;
+import com.dexdrip.stephenblack.nightwatch.DataCollectionService;
+
+import java.util.Date;
+
+/**
+ * Used by DexdripDataReceiver to store new estimates from DexDrip.
+ */
+public class DexDripIntentService extends IntentService {
```
Comment (StephenBlackWasAlreadyTaken) [inline mobile/src/main/java/com/dexdrip/stephenblack/nightwatch/integration/DexdripIntentService.java]: Same with this class

**Comment by StephenBlackWasAlreadyTaken** [comment]
Im still shocked you did this so cleanly, I was all ready to add a new service into dexdrip and the watch app into dexdrip as well

**Comment by bhandfast** [comment]
I'll fix the naming after work today. Funny that my Studio didn't complain, maybe windows is to blame...

**Comment by StephenBlackWasAlreadyTaken** [comment]
Does windows ignore capitalization in file names? I never remember which does which, Im using a mac currently
Anyway, as an update, this is awesome and chugging along just fine on my phone!

**Comment by bhandfast** [comment]
See if it works now. I ended up moving the files to a new package.

**Comment by StephenBlackWasAlreadyTaken** [comment]
Works great!! Thank you!!!

**Pull request merged**